tutorial/managing-software.md

```diff
@@ -0,0 +1,1394 @@
+/lib/systemd
+/lib/systemd/system
+/lib/systemd/system/ssl-cert.service
+/usr
+/usr/sbin
+/usr/sbin/make-ssl-cert
+/usr/share
+/usr/share/doc
+/usr/share/doc/ssl-cert
+/usr/share/doc/ssl-cert/README
+/usr/share/doc/ssl-cert/changelog.gz
+/usr/share/doc/ssl-cert/copyright
+/usr/share/lintian
+/usr/share/lintian/overrides
+/usr/share/lintian/overrides/ssl-cert
+[...]
+```
+
+If we find a file but we're not sure what package it comes from, `dpkg` can
+help us there too! Let's use the example of one of the files from the previous
+output: `/usr/share/ssl-cert/ssleay.cnf` and do a search for it using `dpkg`:
+
+```bash
+dpkg --search /usr/share/ssl-cert/ssleay.cnf
+```
+
+This will provide us with the package name for the given file:
+
+```bash
+ssl-cert: /usr/share/ssl-cert/ssleay.cnf
+```
+
+### Conffiles
+
+Most of a package's configuration is handled through
+[configuration files](https://www.debian.org/doc/debian-policy/ap-pkg-conffiles.html#automatic-handling-of-configuration-files-by-dpkg)
+(often known as **conffiles**).
+
+Package conffiles are different from all other files delivered in a package.
+A package may have any number of conffiles (including none!). Conffiles are
+explicitly marked by the package maintainer during development to protect them
+from being automatically upgraded, whilst all other types of files are not --
+any changes you make to regular files *will be overwritten* during an upgrade. 
+
+### How upgrades are handled
+
+Since a conffile can be changed by us, we might end up with conflicts when the
+package maintainer changes those same files. Therefore, it's important to
+understand how such conflicts are handled.
+
+We can show the four possible upgrade scenarios using the following table. What
+happens during an upgrade depends on whether the conffile on our system has
+been changed by us ("changed/not changed by user"), and whether the default
+content has been changed by the package maintainer ("same as/changed from
+default"):
+
+| The conffile is...          | **same as default**  | **changed from default**  |
+| ---                         | ---                  | ---                       |
+| **...changed by user**      | Keep user's changes  | Ask user                  |
+| **...not changed by user**  | No changes to make   | Apply changes from update |
+
+### Identifying conffiles
+
+Out of the list of files in a package, how do we know which ones are the
+conffiles?
+
+After all, they are not marked by any particular file extension, and
+although they are often found in the `/etc/` directory, they don't *have* to be
+there. As we saw before, the only thing conffiles have in common is that the
+package maintainer decided to mark them as such.
+
+But that's our clue! So once more, `dpkg` can come to our rescue. The
+following command will show us the subset of files in a package that have been
+marked as "conffiles" by the maintainer:
+
+```bash
+dpkg-query --show -f='${Conffiles}\n' apache2
```

Comment:
Can you explain what the parts of this command are doing? So it’s not like “hello beginner just run this magic”

Reply:
Not easily/in a way that doesn't invite more questions, but I've tried :upside_down_face:. I've added the bits of the command to the relevant keywords of the paragraph above to highlight them, but it's still not ideal.

Reply:
I've also added the manpage command, partly so users can investigate the options if they want, but also to get them used to using the manpages. I'm still not 100% happy, but I think it's better than what I originally had.